src/helpers/class-forms.php

```diff
@@ -28,4 +28,46 @@ public static function maybe_override_value_from_query_string( string $value, st
 
     return $value;
   }
+
+  /**
+   * Build a single fast (key based) array for checking which from type(s) is/are used.
+   *
+   * @param  bool   $is_complex                  Is form complex? (uses multiple types).
+   * @param  string $form_type                   Used form type (used if not complex).
+   * @param  array  $form_types_complex          Used form types.
+   * @param  array  $form_types_complex_redirect Used form types that redirect on success.
+   * @return array
+   */
+  public static function detect_used_types( bool $is_complex, string $form_type, array $form_types_complex, array $form_types_complex_redirect ): array {
+    $used_types = [];
+
+    if ( $is_complex ) {
+      foreach ( array_merge( $form_types_complex, $form_types_complex_redirect ) as $complex_form_type ) {
```

Comment:
Eww, this is really not readable 😣

Can you merge arrays outside and then foreach them. The memory impact of assigning an extra variable should be negligible.